src/modules/api-server/__tests__/utils.test.ts

```diff
@@ -0,0 +1,157 @@
+import { Mock, afterAll, describe, beforeEach, expect, it, vi } from 'vitest';
+import { createRequester } from '../utils';
+
+const initialFetch = global.fetch;
+let mockFetch: Mock;
+
+beforeEach(() => {
+  mockFetch = vi.fn();
+  global.fetch = mockFetch;
+});
+afterAll(() => {
+  global.fetch = initialFetch;
```

Comment:
What happens if the fetch isn't "reset"?

Reply:
It could cause other integration tests to fail